src/tests/system/tests/test_local_overrides.py

```diff
@@ -0,0 +1,263 @@
+        5. Local override is deleted
+    :customerscenario: False
+    """
+    provider.group("group1").add(gid=999999)
+    client.sssd.start()
+
+    client.local_override.group("group1").add(name="o-group1")
+    client.sssd.restart()
+
+    assert client.tools.getent.group("group1")
+    assert client.tools.getent.group("o-group1")
+    assert client.tools.getent.group("group1").gid != 888888
+    client.local_override.group("group1").modify(gid=888888
+)
+    assert client.tools.getent.group("group1").gid == 888888
+
+    client.local_override.group("group1").delete()
+    assert client.local_override.group("group1").get() == {}
+
+
+@pytest.mark.topology(KnownTopology.LDAP)
+@pytest.mark.topology(KnownTopology.AD)
+def test_local_overrides__root_user(client: Client, provider: GenericProvider):
+    """
+    :title: Local override root user
+    :setup:
+        1. Start SSSD
+    :steps:
+        1. Create local override "root" for the user1 user
+    :expectedresults:
+        1. Local override is created but the root user contents are not modified
+    :customerscenario: False
+    """
+    provider.user("user1").add(
+        uid=999011, gid=999011, home="/home/user1", gecos="user", shell="/bin/bash", password="Secret123")
+    client.sssd.restart()
+
+    client.local_override.user("user1").add(name="root", uid=999022, gid=999022)
+    assert client.tools.getent.passwd("root").uid == 0
+    assert client.tools.getent.passwd("root").gid == 0
+
+
+@pytest.mark.topology(KnownTopology.LDAP)
+@pytest.mark.topology(KnownTopology.AD)
+def test_local_overrides__nested(client: Client, provider: GenericProvider):
+    """
+    :title: Local override the local override
+    :setup:
+        1. Create posix user "user1" with posix attributes defined
+        2. Start SSSD
+    :steps:
+        1. Override "user1" to "o-user1"
+        2. Override "o-user1" to "lo-user1"
+        3. Authenticate as "user1", short and fully qualified name
+        4. Authenticate as "o-user1", short and fully qualified name
+        5. Authenticate as "lo-user1", short and fully qualified name
+        6. Search for user local overrides
+    :expectedresults:
+        1. User local override is created
+        2. Nested user local override is created
+        3. Authentication successful for both short and fully qualified name
+        4. Authentication successful for both short and fully qualified name
+        5. Authentication successful for both short and fully qualified name
+        6. Local overrides is found for all users
+    :customerscenario: False
+    """
+    provider.user("user1").add(
+        uid=999011, gid=999011, home="/home/user1", gecos="user", shell="/bin/bash", password="Secret123")
+    client.sssd.start()
+    assert client.ssh("user1", "Secret123")
+    client.local_override.user("user1").add(name="o-user1")
+    client.sssd.restart()
+    client.local_override.user("user1").add(name="lo-user1")
```

Comment:
Should this be `o-user1` instead of `user` if it is supposed to be local override of a local override?

This looks like it's just an additional override for user1 and not override for o-user1.

Reply:
Yes indeed.